### PR TITLE
Refactored macros in guidsl to be a bit more readable

### PR DESCRIPTION
**Pull request by beef331**
Hopefully you're fine with the callblock syntax, I personally find it makes macros much easier to write and read, feel free to complain about anything.

### CODE REVIEW

File: owlkettle/guidsl.nim [hunk 1]
```diff
@@ -193,42 +192,33 @@ proc gen(node: Node, stmts, parent: NimNode) =
...
-        bind_sym("true")
-      ))
-      stmts.add(new_assignment(
-        new_dot_expr(parent, "val_" & node.name),
-        node.value
-      ))
+      stmts.add:
+        new_assignment(new_dot_expr(parent, "has_" & node.name), bind_sym("true"))
+      stmts.add:
+        new_assignment(new_dot_expr(parent, "val_" & node.name), node.value)
```
Comment (can-lehmann) [inline owlkettle/guidsl.nim:198]: Maybe use `gen_ast` here as well? I think that it is more readable that way (untested):
```nim
gen_ast(parent, has = ident("has_" & node.name), val = ident("val_" & node.name), value = node.value):
  parent.has = true
  parent.val = value
```

File: owlkettle/guidsl.nim [hunk 2]
```diff
@@ -193,42 +192,33 @@ proc gen(node: Node, stmts, parent: NimNode) =
...
-      stmts.add(new_tree(nnkForStmt, node.vars & @[
-        node.iter,
-        new_stmt_list(
-          new_call(bind_sym("custom_capture"),
-            node.vars.find_variables() & @[body]
-          )
-        )
-      ]))
+      stmts.add:
+        nnkForStmt.newTree:
```
Comment (can-lehmann) [inline owlkettle/guidsl.nim:214]: In this case the regular call (not callblock syntax) also is more readable to me, but I guess that is somewhat subjective.

**Review by can-lehmann — CHANGES_REQUESTED** [review]

File: owlkettle/guidsl.nim [hunk 3]
```diff
@@ -256,7 +246,7 @@ proc gen(node: Node, stmts, parent: NimNode) =
 
 macro gui*(tree: untyped): untyped =
   let gui = tree.parse_gui()
-  result = new_nim_node(nnkStmtListExpr)
+  result = newStmtList()
```
[the remark this answers is not in the document]
Comment (beef331) [reply owlkettle/guidsl.nim]: I hate you :laughing:

**Comment by can-lehmann** [comment]
Looks good now, I will test and merge this in the morning.

**Review by can-lehmann — APPROVED** [review]

File: owlkettle/guidsl.nim (hunk 3 above)
[the remark this answers is not in the document]
Comment (can-lehmann) [reply owlkettle/guidsl.nim]: Oh, thank you :smile: I am sure that you will just love the [style guide](https://github.com/can-lehmann/owlkettle/blob/main/CONTRIBUTING.md#style-guide) I started writing, though it seems like I have to change it to allow callblock syntax for procedures now...

**Pull request merged**